spdk-core/src/backend/backend.rs

```diff
@@ -0,0 +1,33 @@
+use std::ops::RangeInclusive;
+
+use anyhow::Result;
+use bitcoin::{absolute::Height, Amount};
+
+use crate::{BlockData, SpentIndexData, UtxoData};
+
+/// Iterator type for block data that conditionally includes `Send` bound.
+///
+/// - For native targets: includes `Send` bound for thread safety
+/// - For WASM targets: omits `Send` since WASM is single-threaded
+// For native targets, we require Send
+#[cfg(not(target_arch = "wasm32"))]
+pub type BlockDataIterator = Box<dyn Iterator<Item = Result<BlockData>> + Send>;
+
+// For WASM targets, we don't require Send
+#[cfg(target_arch = "wasm32")]
+pub type BlockDataIterator = Box<dyn Iterator<Item = Result<BlockData>>>;
+
+pub trait ChainBackend {
+    fn get_block_data_for_range(
+        &self,
+        range: RangeInclusive<u32>,
+        dust_limit: Amount,
+        with_cutthrough: bool,
+    ) -> BlockDataIterator;
```

Comment:
what about:
```suggestion
pub trait ChainBackend {
    fn get_block_data_for_range(
        &self,
        range: RangeInclusive<u32>,
        dust_limit: Option<Amount>,
        with_cutthrough: bool,
    ) -> Result<BlockDataIterator, ()>;
```

I'm mainly thinking if a backend do not support dust_limit/cutthrough

Reply:
yes, and also if the backend dont support cutthrough